### PR TITLE
[python] fixes for supporting 2d numpy arrays for predictions, grads and hess in multiclass custom objective and eval

### PR DESCRIPTION
**Pull request by StrikerRUS**
I hadn't had a chance to review #4925, so I'm proposing some fixes and my suggestions in a form of a separate follow-up PR. Sorry about that.

### CODE REVIEW

File: python-package/lightgbm/sklearn.py [hunk 1]
```diff
@@ -72,13 +72,13 @@ def __init__(self, func: _LGBM_ScikitCustomObjectiveFunction):
...
                     with respect to the elements of y_pred for each sample point.
-                hess : numpy 1-D array of shape = [n_samples] or numpy 2-D array of shape [n_samples, n_classes] (for multi-class task)
+                hess : numpy 1-D array of shape = [n_samples] or numpy 2-D array of shape = [n_samples, n_classes] (for multi-class task)
                     The value of the second order derivative (Hessian) of the loss
                     with respect to the elements of y_pred for each sample point.
 
         .. note::
 
-            For multi-class task, preds are a [n_samples, n_classes] numpy 2-D array,
+            For multi-class task, y_pred are numpy 2-D array of shape = [n_samples, n_classes],
```
Comment (jmoralez) [inline python-package/lightgbm/sklearn.py:81]:
```suggestion
            For multi-class task, y_pred is numpy 2-D array of shape = [n_samples, n_classes],
```

File: python-package/lightgbm/sklearn.py [hunk 2]
```diff
@@ -473,7 +465,7 @@ def __init__(
                 The value of the second order derivative (Hessian) of the loss
                 with respect to the elements of y_pred for each sample point.
 
-        For multi-class task, preds are a [n_samples, n_classes] numpy 2-D array,
+        For multi-class task, y_pred are numpy 2-D array of shape = [n_samples, n_classes],
```
Comment (jmoralez) [inline python-package/lightgbm/sklearn.py:468]:
```suggestion
        For multi-class task, y_pred is numpy 2-D array of shape = [n_samples, n_classes],
```

**Review by jmoralez — COMMENTED** [review]
Thank you for this! I just left some suggestions on the wording for `y_pred`

**Comment by shiyu1994** [comment]
@StrikerRUS @jmoralez Thank you for the fix! I'll be more careful next time when reviewing a python PR.

**Review by jmoralez — APPROVED** [review]

**Pull request merged**